src/shogun/util/factory.h

```diff
@@ -333,5 +335,63 @@ namespace shogun
 	{
 		return std::make_shared<PipelineBuilder>();
 	}
+
+	class StringsFeatures;
+	class FeatureSubset;
+
+	template <
+	    typename TypeName, typename... Args,
+	    std::enable_if_t<
+	        traits::is_any_of_v<
+	            TypeName, Features, StringsFeatures, FeatureSubset,
+	            FeatureSubset, CSVFile, LibSVMFile, PipelineBuilder, Kernel>,
+	        TypeName>* = nullptr>
+	std::shared_ptr<TypeName> create(Args&&... args)
+	{
+		if constexpr (std::is_same_v<TypeName, Features>)
+		{
+			return features(std::forward<Args>(args)...);
```

Comment:
so then all these "old" methods will be hidden into somewhere and not be exposed anymore I guess?

Reply:
ah this is the comment above by the others, nevermind :)